aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -257,6 +257,28 @@ CREATE_UNARY_UFUNC_NO_INPLACE(signbit);
 CREATE_UNARY_UFUNC_FUNCTIONAL(isnan);
 CREATE_UNARY_UFUNC_FUNCTIONAL(isinf);
 
+// pow function for Sparse CSR, since square is composite - this will also enable square for Sparse CSR
+Tensor& pow_sparse_csr_scalar_out(const Tensor& self, const Scalar& exponent, Tensor& result) {
+  TORCH_CHECK(!exponent.isComplex() && !self.is_complex(), "Complex Exponent is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(!(exponent.isBoolean() && exponent.equal(false)), "Exponent as False is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(exponent.to<double>() > 0.0, "Exponent must be greater than 0 for Sparse CSR Layout.");
+  return unary_op_out<Tensor& (*)(const Tensor&, const Scalar&, Tensor&)>(&at::pow_outf, self, result, exponent);
+}
+
+Tensor pow_sparse_csr_scalar(const Tensor& self, const Scalar& exponent) {
+  TORCH_CHECK(!exponent.isComplex() && !self.is_complex(), "Complex Exponent is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(!(exponent.isBoolean() && exponent.equal(false)), "Exponent as False is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(exponent.to<double>() > 0.0, "Exponent must be greater than 0 for Sparse CSR Layout.");
+  return get_result_tensor_for_unary_op<Tensor (*)(const Tensor&, const Scalar&)>(&at::pow, self, exponent);
+}
+
+Tensor& pow_sparse_csr_scalar_(Tensor& self, const Scalar& exponent) {
+  TORCH_CHECK(!exponent.isComplex() && !self.is_complex(), "Complex Exponent is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(!(exponent.isBoolean() && exponent.equal(false)), "Exponent as False is not supported for Sparse CSR Layout.");
+  TORCH_CHECK(exponent.to<double>() > 0.0, "Exponent must be greater than 0 for Sparse CSR Layout.");
+  return unary_op_inplace(self, &Tensor::pow_, exponent);
```

Comment:
Hi, @IvanYashchuk - Can you please help here? I'm getting the following error while using `&Tensor::pow_`:

```cpp
../aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp: In function ‘at::Tensor& at::native::pow_sparse_csr_scalar_(at::Tensor&, const c10::Scalar&)’:
../aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:279:56: error: no matching function for call to ‘unary_op_inplace(at::Tensor&, <unresolved overloaded function type>, const c10::Scalar&)’
   return unary_op_inplace(self, &Tensor::pow_, exponent);
                                                        ^
../aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:100:9: note: candidate: template<class F, class ... Args> at::Tensor& at::{anonymous}::unary_op_inplace(at::Tensor&, const F&, Args&& ...)
 Tensor& unary_op_inplace(Tensor& self, const F& op_inplace, Args&&... args) {
         ^
../aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:100:9: note:   template argument deduction/substitution failed:
../aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:279:56: note:   couldn't deduce template parameter ‘F’
   return unary_op_inplace(self, &Tensor::pow_, exponent);
```

I understand it can be because of `pow_` having overloaded alternatives (`(Tensor, Tensor), (Tensor, Scalar)`) - I tried explicitly passing the type to the template `unary_op_inplace`, but that also fails.

```cpp
return unary_op_inplace<Tensor& (*)(const Scalar&)>(self, &Tensor::pow_, exponent);
```

(fails with `no matching function call to`... error)

Reply:
Let's use here `self.values().pow_(exponent)` instead of a templated helper function.